### PR TITLE
chore: bump opendal version to 0.49

### PR DESCRIPTION
**Pull request by WenyXu**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## What's changed and what's your intention?

bump opendal version to 0.49

## Checklist

- [ ] I have written the necessary rustdoc comments.
- [ ] I have added the necessary unit tests and integration tests.
- [ ] This PR requires documentation updates.

### CODE REVIEW

File: src/object-store/src/util.rs [hunk 1]
```diff
@@ -140,17 +142,106 @@ pub(crate) fn extract_parent_path(path: &str) -> &str {
...
+                    format_args!(
+                        "{}",
+                        context.iter().enumerate().map(|(i, (k, v))| {
+                            if i > 0 {
+                                format!(" {}={}", k, v)
+                            } else {
+                                format!("{}={}", k, v)
+                            }
+                        }).collect::<String>()
+                    ),
```
Comment (evenyag) [inline src/object-store/src/util.rs:182]: The implementation is updated, you can refer to this PR
https://github.com/apache/opendal/pull/5021/

File: src/object-store/src/util.rs [hunk 2]
```diff
@@ -140,17 +142,106 @@ pub(crate) fn extract_parent_path(path: &str) -> &str {
...
+                    format!("{err:?}")
+                );
+            };
+        }
+
+        // Print debug message if operation is oneshot, otherwise in trace.
+        if operation.is_oneshot() {
```
Comment (evenyag) [inline src/object-store/src/util.rs:207]: You can use the `log!` macro and specify the level according to the operation.

File: src/object-store/src/util.rs [hunk 3]
```diff
@@ -140,17 +142,106 @@ pub(crate) fn extract_parent_path(path: &str) -> &str {
...
+                        "{}",
+                        context.iter().enumerate().map(|(i, (k, v))| {
+                            if i > 0 {
+                                format!(" {}={}", k, v)
+                            } else {
+                                format!("{}={}", k, v)
+                            }
+                        }).collect::<String>()
+                    ),
+                    format!("{err:?}")
```
Comment (evenyag) [inline src/object-store/src/util.rs:183]: For unexpected errors, we may need to use `:#?`.

File: src/object-store/src/util.rs [hunk 4]
```diff
@@ -140,17 +142,106 @@ pub(crate) fn extract_parent_path(path: &str) -> &str {
...
+        context: &[(&str, &str)],
+        message: &str,
+        err: Option<&opendal::Error>,
+    ) {
+        if let Some(err) = err {
+            // Print error if it's unexpected, otherwise in debug.
+            if err.kind() == ErrorKind::Unexpected {
+                error!(
+                    target: LOGGING_TARGET,
+                    "service={} name={} {}: {operation} {message} {}",
```
Comment (evenyag) [inline src/object-store/src/util.rs:170]: `format!("{err}")` is unnecessary.
```suggestion
                    "service={} name={} {}: {operation} {message} {err:#?}",
```

File: src/object-store/src/util.rs [hunk 5]
```diff
@@ -140,17 +144,83 @@ pub(crate) fn extract_parent_path(path: &str) -> &str {
...
+    fn log(
+        &self,
+        info: &AccessorInfo,
+        operation: Operation,
+        context: &[(&str, &str)],
+        message: &str,
+        err: Option<&opendal::Error>,
+    ) {
+        if let Some(err) = err {
+            // Print error if it's unexpected, otherwise in warn.
```
Comment (evenyag) [inline src/object-store/src/util.rs:183]:
```suggestion
            // Print error if it's unexpected, otherwise in error.
```

**Review by evenyag — APPROVED** [review]

**Review by waynexia — APPROVED** [review]

**Pull request merged**